### PR TITLE
Product gallery - fix “View all” link

### LINKED ISSUE #11988: Product Gallery: View all link sizing
When the thumbnails get smaller due to increased Number of Thumbnails setting, the "View All" link text becomes too big and shows outside of the bounding box.

### PR DESCRIPTION
**Pull request by roykho**
## What

Fixes https://github.com/woocommerce/woocommerce-blocks/issues/11988

This PR adds specific styling to the "View All" link and thumbnails count depending on how many thumbnails are displayed.

- When set to 3 - 5 thumbnails both thumbnail count and "View All" link is shown.
- When set to 6 - 8 thumbnails only count is shown because "View All" link doesn't have enough space to display nicely.

## Why

This is so that shoppers will have a better visual experience and not have the "View All" link out of place or not showing completely.

## Testing Instructions

_Please consider any edge cases this change may have, and also other areas of the product this may impact._

1. Go to Appearance->Editor->Templates->Single Product.
2. Add the Product Gallery block.
3. Go to a product in the frontend with 8 gallery images added.
4. Ensure you see both the "Count" and "View All" link and are displayed nicely.
5. Go back to the Editor Product Gallery block and change to show 8 thumbnails.
6. Go back to the frontend product with 8 gallery images.
7. Ensure you see only the "Count" being displayed and that it is showing nicely and centered.

* [x] Do not include in the Testing Notes 
* [ ] Should be tested by the development team exclusively 

## Screenshots or screencast

| Before | After |
| ------ | ----- |
|        |       |

## WooCommerce Visibility

Required:

* [ ] WooCommerce Core
* [ ] Feature plugin
* [x] Experimental
* [ ] N/A

## Checklist

Required:

* [x] This PR has either a `[type]` label or a `[skip-changelog]` label.
* [ ] This PR is assigned to a milestone.

Conditional:

* [ ] This PR has a UI change and has been cross-browser tested at different viewport sizes on both the frontend and in the editor.
* [ ] This PR has a changelog description (if `[skip-changelog]` label is not present).
* [ ] This PR adds/removes a feature flag & I've updated [this doc](https://github.com/woocommerce/woocommerce-blocks/blob/trunk/docs/internal-developers/blocks/feature-flags-and-experimental-interfaces.md).
* [ ] This PR adds/removes an experimental interfaces, and I've updated [this doc](https://github.com/woocommerce/woocommerce-blocks/blob/trunk/docs/internal-developers/blocks/feature-flags-and-experimental-interfaces.md).
* [ ] This PR has been accessibility tested.
* [ ] This PR has had any necessary documentation added/updated.

## Changelog

> Add suggested changelog entry here.

### CODE REVIEW

File: src/BlockTypes/ProductGalleryThumbnails.php [hunk 1]
```diff
@@ -170,13 +170,14 @@ protected function render( $attributes, $content, $block ) {
 					}
 
 					return sprintf(
-						'<div class="wc-block-product-gallery-thumbnails wp-block-woocommerce-product-gallery-thumbnails %1$s" style="%2$s" data-wc-interactive=\'%4$s\'>
-							%3$s
+						'<div class="wc-block-product-gallery-thumbnails wp-block-woocommerce-product-gallery-thumbnails %1$s wc-block-product-gallery-thumbnails__total-count-%2$s" style="%3$s" data-wc-interactive=\'%4$s\'>
```
Comment (danieldudzic) [inline src/BlockTypes/ProductGalleryThumbnails.php]: I don't think it's necessary to add this additional class for tracking the number of thumbnails.

You can utilize the attribute that's on the parent Product Gallery element.

```
		#{$gallery}[data-thumbnails-number-of-thumbnails='4'] & {

		}
```

Note: this class is not present when the number of thumbnails is default, that is `3`.

File: assets/js/blocks/product-gallery/style.scss [hunk 2]
```diff
@@ -320,6 +320,30 @@ $default-number-of-thumbnails: 3;
...
+	&.wc-block-product-gallery-thumbnails__total-count-7,
+	&.wc-block-product-gallery-thumbnails__total-count-8 {
+		.wc-block-product-gallery-thumbnails__thumbnail__remaining-thumbnails-count {
+			@include font-size(small);
+		}
+
+		.wc-block-product-gallery-thumbnails__thumbnail__view-all {
+			display: none;
+		}
+	}
```
Comment (danieldudzic) [inline assets/js/blocks/product-gallery/style.scss]: You can use a loop to make the code a bit more concise:

```
	@for $i from 3 through 8 {
		#{$gallery}[data-thumbnails-number-of-thumbnails='#{$i}'] & {

		}
	}
```

**Review by danieldudzic — CHANGES_REQUESTED** [review]
I'd rethink the sizing of the View all overlay text.

Is there a way we can make the font-size a bit more fluid?

The Product Gallery might be used for larger widths, and that might result with a very small `View all` text with a static font size:

Moto_Guzzi_v7_–_newproductgallery

**Comment by danieldudzic** [comment]
At request from @roykho I'm taking over this PR.

**Comment by danieldudzic** [comment]
@roykho Would you mind reviewing? 🙏

**Review by roykho — COMMENTED** [review]
@danieldudzic changes looks good!

**Review by danieldudzic — APPROVED** [review]

**Pull request merged**